src/Simplex/Chat/Migrations/M20240102_note_folders.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Simplex.Chat.Migrations.M20240102_note_folders where
+
+import Database.SQLite.Simple (Query)
+import Database.SQLite.Simple.QQ (sql)
+
+m20240102_note_folders :: Query
+m20240102_note_folders =
+  [sql|
+    CREATE TABLE note_folders (
+      note_folder_id INTEGER PRIMARY KEY AUTOINCREMENT,
+      user_id INTEGER NOT NULL REFERENCES users ON DELETE CASCADE,
+      created_at TEXT NOT NULL DEFAULT(datetime('now')),
+      updated_at TEXT NOT NULL DEFAULT(datetime('now')),
+      chat_ts TEXT NOT NULL DEFAULT(datetime('now')),
+      favorite INTEGER NOT NULL DEFAULT 0,
+      unread_chat INTEGER NOT NULL DEFAULT 0
+    );
+
+    ALTER TABLE chat_items ADD COLUMN note_folder_id INTEGER DEFAULT NULL REFERENCES note_folders ON DELETE CASCADE;
+    ALTER TABLE files ADD COLUMN note_folder_id INTEGER DEFAULT NULL REFERENCES note_folders ON DELETE CASCADE;
+
+    CREATE INDEX chat_items_note_folder_id ON chat_items(note_folder_id);
+    CREATE INDEX files_note_folder_id ON files(note_folder_id);
+    CREATE UNIQUE INDEX note_folders_user_id ON note_folders(user_id); -- drop UNIQUE for more folders per user
```

Comment:
non unique was ok, also no need to check constraint in query - we plan to add more note chats soon (tips)
```suggestion
    CREATE INDEX note_folders_user_id ON note_folders(user_id);
```

Reply:
But it *is* unique. And this check caught me a few bugs already.

Reply:
It's single for now but making it unique seems wrong - we'll soon add tips and user would have multiple local chats. So it's not the record in table that would be one per user making user_id unique, but this specific type of local chats (notes) that would, but we don't even have the field to differentiate them by yet. That some functions had bugs now doesn't mean we should keep constraint in schema and plan to recreate restrictive index later.

Reply:
ok, moved to explicit check in `createNoteFolder`.